modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/capability/xml/schema/builder/SchemaBuilder.java

```diff
@@ -694,10 +716,10 @@ private DataQualifierVisitor getParameterDeclarationVisitor(final ExtensionType
             @Override
             public void onList()
             {
-                forceOptional = true;
+                forceOptional = shouldForceOptional();
                 defaultOperation();
                 DataType genericType = parameterModel.getType().getGenericTypes()[0];
-                if (genericType != null && shouldGenerateListChildElements(genericType))
+                if (genericType != null && shouldGenerateDataTypeChildElements(genericType) && !parameterModel.getExpressionSupport().equals(ExpressionSupport.REQUIRED))
```

Comment:
this long line is hard to read and repeated below. Wouldn't it be better becomes responsible for checking that genericType is != null and that the extensionSupport is not REQUIRED?